app/controllers/api/application_controller.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+# rubocop:disable ClassAndModuleChildren
+class API::ApplicationController < ApplicationController
+  include Rails::Pagination
+
+  before_action :add_security_headers
+
+  def authenticate_user!
+    return become_active if logged_in? && adequate_scopes? && current_user.authorized_access_token?
+    render_forbidden
+  end
+
+  private
+
+  def render_forbidden
+    render json: {
+      message: "Unauthorized"
+    }, status: :forbidden
+  end
+
+  def add_security_headers
```

Comment:
Send security headers so we can access Pagination headers

Reply:
I'll admit I'm not super well versed in Access-Control headers in Rails. This seems fine, but @mastahyeti would you mind :eyes: this?

Reply:
What is going to be making XHR requests to this? This allows any origin to make a GET request to any URL, sending any headers. It seems like that is _probably_ more open than necessary, but I'm not sure how this API is being used.

Reply:
Classroom-Desktop is making requests to this API to get student's assignment repos. I figured the pagination headers were ok to expose to everyone because they didn't have any sensitive data? What do you think?

Reply:
A desktop app isn't going to be bound by CORS and won't have any restrictions on what headers it can read. CORS defines hows one website is allowed to access resources from another website and is enforced by web browsers. If this API is only being used by a native app you probably don't need to make any CORS changes.

Reply:
Was running into problems with CORS since Electron technically makes a request from a different origin which is why I added these headers originally, but I think I found a way around that while investigating the problem again.

I can turn off the same origin policy for the browser window by setting the `webSecurity` in the browser window options to false (https://github.com/electron/electron/blob/master/docs/api/browser-window.md#new-browserwindowoptions).

Is that a safer approach or should I keep `webSecurity` on and expose the response headers?

Reply:
What JavaScript networking APIs are you using in the Electron app? You should be able to use the [`net`](https://electronjs.org/docs/api/net) API instead of trying to make XHR requests.

Reply:
@mastahyeti and I just discussed different options for authentication here, and agree that this will likely need to be converted to a token based API in a future PR. I can't use the `net` api since I'm currently using the browser session for authentication on the Classroom-Desktop app.

For this PR, I'm going to keep the CORS headers in so we can access the API from Classroom-Desktop without turning off `webSecurity` in Electron (as that poses other security risks).  I've created an issue on the appsec review to get more 👀 on this: https://github.com/github/appsec-reviews/issues/230